src/idl_gen_go.cpp

```diff
@@ -831,19 +914,26 @@ class GoGenerator : public BaseGenerator {
         }
       }
       code += ")\n\n";
+    } else {
+      if (is_enum) {
+        code += "import \"strconv\"\n\n";
+      }
     }
   }
 
   // Save out the generated code for a Go Table type.
   bool SaveType(const Definition &def, const std::string &classcode,
-                bool needs_imports) {
+                const bool needs_imports, const bool is_enum) {
     if (!classcode.length()) return true;
 
     Namespace &ns = go_namespace_.components.empty() ? *def.defined_namespace
                                                      : go_namespace_;
     std::string code = "";
-    BeginFile(LastNamespacePart(ns), needs_imports, &code);
+    BeginFile(LastNamespacePart(ns), needs_imports, is_enum, &code);
     code += classcode;
+    while (code.substr(code.length() - 2) == "\n\n") {
```

Comment:
`code` can have length zero, and `length` would return an unsigned int, which could cause an unchecked integer underflow error. It's unlikely, but even so, this code seems tricky and potentially hard to maintain. 

Is there a way to do what you need in some simpler way? Ideally without using `pop_back`?

Reply:
BeginFile always adds much more than two characters, and can't ever not do that, because any valid Go file must contain at least `package`. But a defensive check seems prudent. The simplest option is just to write
```
while (code.length() > 2 && code.substr(code.length() - 2) == "\n\n")
```
Do you have a better way to do it? The goal, of course, is to strip extra newlines at the end of the file to make it gofmt-clean.

Reply:
@aardappel WDYT about this C++ question?

Reply:
Nevermind, this is an outdated diff.